### PR TITLE
[docs] Remove TypeScript v3 outdated version mentions

### LINKED ISSUE #14115: [data grid][docs] Improve theme augmentation documentation
### Latest version

- [X] I have tested the latest version

### Steps to reproduce

[Link to live example: (required)](https://mui.com/x/react-data-grid/style/)

Steps:
import { createTheme } from '@mui/material/styles';

```
const theme = createTheme({
  mixins: {
    MuiDataGrid: {
      // Pinned columns sections
      pinnedBackground: '#340606',
      // Headers, and top & bottom fixed rows
      containerBackground: '#343434',
    },
  },
});
```
The createTheme does not contain a field for MuiDataGrid

### Current behavior

The createTheme does not contain a field for MuiDataGrid

### Expected behavior

intellisense enable and field available

### Context

```
import { createTheme } from '@mui/material/styles';

const theme = createTheme({
  mixins: {
    MuiDataGrid: {
      // Pinned columns sections
      pinnedBackground: '#340606',
      // Headers, and top & bottom fixed rows
      containerBackground: '#343434',
    },
  },
});

```
MuiDataGrid field is not enable nor available

### Your environment

    "@mui/material": "^5.16.6",
    "@mui/styled-engine-sc": "^6.0.0-alpha.18",
    "@mui/x-data-grid": "^7.12.0",
    "@emotion/react": "^11.13.0",
    "@emotion/styled": "^11.13.0",

**Search keywords**: MuiDataGrid

### LINKED ISSUE #14239: [docs] Remove TypeScript v3 outdated version mentions
### Steps to reproduce

We have a couple of mentions of TypeScript v3 https://github.com/search?q=org%3Amui+%22TypeScript+3.x%22&type=code. But per our version policy, following DefinitelyTypes we should remove them, we don't support this anymore see https://github.com/mui/material-ui/issues/40996 

### Context

First raised in https://github.com/mui/mui-x/issues/14115#issuecomment-2269980017. 

### Your environment

<details>
  <summary><code>npx @mui/envinfo</code></summary>

```
  Don't forget to mention which browser you used.
  Output from `npx @mui/envinfo` goes here.
```
</details>

**Search keywords**: -

### PR DESCRIPTION
**Pull request by k-rajat19**
MateriaL UI sets the minimum supported ts version to 4.7, so it's time to remove v3 mentions in docs.
Fixes #14239

### CODE REVIEW

File: docs/data/data-grid/style/style.md [hunk 1]
```diff
@@ -122,6 +122,7 @@ By default, the Data Grid uses the Material UI `theme.palette.background.defaul
 
 ```tsx
 import { createTheme } from '@mui/material/styles';
+import type {} from "@mui/x-data-grid/themeAugmentation";
```
Comment (k-rajat19) [inline docs/data/data-grid/style/style.md:125]: this fixes #14115, though it is already mentioned here https://mui.com/x/react-data-grid/getting-started/#typescript

**Review by romgrk — APPROVED** [review]

**Review by LukasTy — APPROVED** [review]
Thank you for your contribution! 🙏

**Comment by k-rajat19** [comment]
hm...same picker test is failing again, it seems like https://github.com/mui/mui-x/pull/14457 missed something ?

**Comment by LukasTy** [comment]
> hm...same picker test is failing again, it seems like #14457 missed something ?

This branch was just out of sync with `upstream/master`.
I synced it and it should be great now. 👍

File: docs/data/data-grid/style/style.md (hunk 1 above)
Comment (oliviertassinari) [reply docs/data/data-grid/style/style.md]: For #14115, I don't think that this PR solves it, IMHO, https://github.com/mui/mui-x/issues/14115#issuecomment-2271923988

Comment (k-rajat19) [reply docs/data/data-grid/style/style.md]: will do this in a separate PR 👍

**Pull request merged**